core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java

```diff
@@ -235,7 +235,7 @@ private void performRewrite(List<ManifestFile> currentManifests) {
               keptManifests.add(manifest);
             } else {
               rewrittenManifests.add(manifest);
-              try (ManifestReader reader = ManifestFiles.read(manifest, ops.io(), ops.current().specsById())
+              try (ManifestReader<DataFile> reader = ManifestFiles.read(manifest, ops.io(), ops.current().specsById())
```

Comment:
Does this mean we can only do rewrites if there are no `DeleteFile`'s?

Reply:
Never mind, I see that the rewrite is only operating on the data manifests and not all manifests.